trpcultivate_phenotypes/src/Plugin/Validators/Genus.php

```diff
@@ -0,0 +1,135 @@
+      $container->get('trpcultivate_phenotypes.genus_ontology')
+    );
+  }
+
+  /**
+   * Validate items in the phenotypic data upload assets.
+   *
+   * @return array
+   *   An associative array with the following keys.
+   *   - title: string, section or title of the validation as it appears in the result window.
+   *   - status: string, pass if it passed the validation check/test, fail string otherwise and todo string if validation was not applied.
+   *   - details: details about the offending field/value.
+   */
+  public function validate() {
+    // Validate ...
+    $validator_status = [
+      'title' => 'Genus Exists and/or Matches the Project/Experiment',
+      'status' => 'pass',
+      'details' => ''
+    ];
+
+    // Instructed to skip this validation. This will set this validator as upcoming or todo.
+    // This happens when other prior validation failed and this validation could only proceed
+    // when input values in the failed validator have been rectified.  
+    if ($this->skip) {
+      $validator_status['status'] = 'todo';
+      return $validator_status;
+    }
+
+    // Genus:
+    //   - Is not empty
+    //   - Is active, therefore it exits in chado.organism.
+    //   - Matches the set genus for the project
+
+    if (empty($this->genus)) {
+      $validator_status['status']  = 'fail';
+      $validator_status['details'] = 'Genus field is empty. Please enter a value and try again.';
+    }
+    else {
+      // Test if the genus is an active genus. This will at the same time 
+      // confirm that it exists in organism table.
+      $genus_config = $this->service_genus_ontology->getGenusOntologyConfigValues($this->genus);
+      if (!$genus_config) {
+        // Genus is not recognized by the module.
+        $validator_status['status']  = 'fail';
+        $validator_status['details'] = 'Genus does not exist. Please enter a value and try again.';
+      }
+      else {
+        if ($genus_config['trait'] <= 0) {
+          // Genus exits, but not configured.
+          $validator_status['status']  = 'fail';
+          $validator_status['details'] = 'Genus is not configured. Please enter a value and try again.';
+        }        
+      }
+      
+      // This check should factor in the status of the project entered to see
+      // if that project has genus set or not. The former should check if they match
+      // and the latter should allow so that further in the process will create 
+      // the project-genus relationship.
+
+      // Additional check if genus is pared with project, ensure that
+      // it is the genus the project is set to.
+      if (isset($this->project) && $this->project > 0) {
+        // Check if it has a genus set.
+        $project_id = ChadoProjectAutocompleteController::getProjectId($this->project);
+        $genus = $this->service_genus_project->getGenusOfProject($project_id);
+
+        if ($genus && $this->genus != $genus['genus']) {
```

Comment:
This doesn't take into account our discussion that if the project does not have a genus set at all then it can be assigned here and it not be an error.

So I think you want an `elseif (!$genus)` where you create the genus/project connection 🤔

Reply:
I've updated the project validator not to trigger failed validation if project has no genus set...this will allow for project without genus set be selected. The genus validator will only check if the genus selected is an active genus. Together will allow unset project to select an active genus and further in the import will create the relationship